python/lsst/ts/audiotrigger/read_serial_temp_scanner.py

```diff
@@ -0,0 +1,313 @@
+
+    def config(self):
+        """Configure the temp_scanner."""
+        # TODO: read config .yaml instead
+        PORT = "/dev/ttyUSB0"
+        BAUDRATE = 19200
+        self.sensor_dict = OrderedDict(
+            {
+                "C01": "Ambient",
+                "C02": "Laser",
+                "C03": "FC",
+                "C04": "A",
+                "C05": "B",
+                "C06": "C",
+                "C07": "D",
+                "C08": "E",
+            }
+        )  # These will need to be in some configuration file
+
+        # Define serial connection
+        if self.simulation_mode:
+            self.serial = MockThermalSensor()
+        else:
+            self.serial = serial.Serial(
+                port=PORT,
+                baudrate=BAUDRATE,
+                parity=serial.PARITY_NONE,
+                stopbits=serial.STOPBITS_ONE,
+                bytesize=serial.EIGHTBITS,
+                timeout=1,
+            )
+
+        # fan sensor
+        self.fan_sensor = "Ambient"
+        self.fan_gpio = 4
+        self.fan_turn_on_temp = 25
+        hysteresis = 2
+        self.fan_turn_off_temp = self.fan_turn_on_temp - hysteresis
+        self.configured = True
+
+    async def start(self, **kwargs):
+        """Start reading the temperature channels."""
+        self.task = asyncio.ensure_future(self.serial_temperature_task())
+        await super().start(kwargs=kwargs)
+
+    async def write_if_connected(self, string_to_write):
+        """Write message if connected."""
+        if self.simulation_mode:
+            return
+        if self.connected:
+            await self.write_str(string_to_write)
+
+    async def set_fan(self, setting):
+        """Set the fan on/off.
+
+        Parameters
+        ----------
+        setting : `int`
+            Turn the fan on or off.
+        """
+        if not self.configured or not self.pi.connected:
+            await self.write_if_connected(
+                "TS: Error: Not configured properly before actuating fan"
+            )
+            raise ValueError("Not configured properly before actuating fan")
+        await self.loop.run_in_executor(
+            None, functools.partial(self.pi.write, self.fan_gpio, setting)
+        )
+
+    async def set_fan_on(self):
+        """Set the fan on."""
+        await self.set_fan(Fan.ON)
+
+    async def set_fan_off(self):
+        """Set the fan off."""
+        await self.set_fan(Fan.OFF)
+
+    async def handle_data(self, new_data):
+        """Handle the incoming data.
+
+        Parameters
+        ----------
+        new_data : `dict`
+            New data.
+        """
+        # We only care about one sensor's reading for operating the fan
+        try:
+            new_data[self.fan_sensor] = float(new_data[self.fan_sensor])
+        except Exception as e:
+            await self.write_if_connected(
+                f"TS: Exception trying to convert data to int... Data: {new_data[self.fan_sensor], {str(e)}}"
```

Comment:
If no `self.fan_sensor` key exists, logging will fail. Please take care of that situation.